concourse/steps/draft_release.mako

```diff
@@ -46,9 +52,16 @@ processed_version = version.process_version(
 )
 
 repo_dir = ci.util.existing_dir('${repo.resource_name()}')
-component_descriptor_v2 = cm.ComponentDescriptor.from_dict(
-    ci.util.parse_yaml_file('${component_descriptor_v2_path}'),
-)
+
+try:
+    component_descriptor_v2 = cnudie.util.determine_main_component(
+        repository_hostname='${repo.repo_hostname()}',
+        repository_path='${repo.repo_path()}',
+        component_descriptor_v2_path='${component_descriptor_v2_path}',
+        ctf_path='${ctf_path}',
+    )
+except ValueError as err:
+    ci.util.fail(str(err))
```

Comment:
nit: since you fail anyway, you might omit the try-except (printing the stacktrace helps a lot w/ debugging)